src/day7.scala

```diff
@@ -0,0 +1,74 @@
+// using scala 3.1.0
+// using lib com.lihaoyi::pprint:0.6.6
+
+package day7
+
+import scala.util.Using
+import scala.io.Source
+import scala.annotation.tailrec
+
+case class Crabmarine(
+    horizontal: Int,
+    increasingCost: Boolean,
```

Comment:
Design-wise, I wonder if instead of having a field `increasingCost` we should have two specialized classes with different implementations for `moveUp` and `moveDown`?

Reply:
Refactored it, looks much nicer :100: